connectors/src/connectors/notion/lib/parents.ts

```diff
@@ -162,16 +157,15 @@ export async function updateAllParentsFields(
   // can be desired or not depending on the use case
   const q = new PQueue({ concurrency: 16 });
   const promises: Promise<void>[] = [];
-  for (const pageId of pageIdsToUpdate) {
+  [...pageIdsToUpdate].forEach((pageId, index) => {
```

Comment:
Why this change ? you can `for (const [i, pageId] of pageIdsToUpdate.entries())` instead of copying an array and using a cursed forEach :p